src/main/java/com/github/onsdigital/babbage/api/util/SearchUtils.java

```diff
@@ -73,64 +62,60 @@
 public class SearchUtils {
 
     private static final String DEPARTMENTS_INDEX = "departments";
-    private static final String ERRORS_KEY = "errors";
 
     /**
      * Performs search for requested search term against filtered content types and counts contents types.
      * Content results are serialised into json with key "result" and document counts are serialised as "counts".
      * <p>
      * Accepts extra searches to perform along with content search and document counts.
      *
+     * @param searchTerm
+     * @param queries
+     * @param searchDepartments
+     * @return
+     */
+    public static LinkedHashMap<String, SearchResult> search(String searchTerm, SearchQueries queries, boolean searchDepartments) throws IOException {
+        if (searchTerm == null) {
+            return null;
+        }
+        LinkedHashMap<String, SearchResult> results = searchAll(queries);
+
+        if (searchDepartments) {
+            searchDeparments(searchTerm, results);
+        }
+        return results;
+    }
+
+    /**
+     * Search time series for given search term and if found return the URI to redirect to.
+     *
      * @param request
+     * @param searchTerm
      * @return
      */
-    public static BabbageResponse search(HttpServletRequest request, String listType, String searchTerm, SearchQueries queries, boolean searchDepartments) throws IOException {
+    public static String getRedirectWhenTimeSeries(HttpServletRequest request, String searchTerm) {
+        String redirectUri = null;
         if (searchTerm == null) {
-            return buildResponse(request, listType, null);
+            return null;
         } else if (!isFiltered(request)) { //only search for time series when new search made through search input
             //search time series by cdid, redirect to time series page if found
             String timeSeriesUri = searchTimeSeriesUri(searchTerm);
             if (timeSeriesUri != null) {
+                redirectUri = timeSeriesUri;
```

Comment:
I appreciate that this was here before but _clean the campsite n all that..._ 

Im not a fan of multiple nested conditions in a single method I find it makes the code much harder to read. It also makes the scope of the code harder to understand - each nested block is a specific case to handle. 

Consider pulling the inner conditionals out into their own method and calling them as required from a _master_ method. The top level method then reads like a set of instructions and the specific details of each step is in each method.

Reply:
The inner conditional was completely unnecessary anyway so I simply removed it.